ambari-metrics/ambari-metrics-grafana/ambari-metrics/datasource.js

```diff
@@ -875,7 +883,10 @@ define([
             var componentName = interpolated.substring(0,interpolated.indexOf('.'));
             return this.initMetricAppidMapping()
                 .then(function () {
-                  var storm = allMetrics["nimbus"];
+                  var storm = [];
+                  if(allMetrics["nimbus"]) {
+                    storm = allMetrics["nimbus"];
+                  }
```

Comment:
We could define a method for all those similar occurrences and just use something like `var storm = getMetrics(allMetrics, "nimbus");`